Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,154 @@ sealed class KISAddonConfig : MonoBehaviour {
+      }
     }
+  }
+
+  class KISPodInventoryLoader: LoadingSystem
+  {
+    public bool done;
 
-    // Kerbal parts.
-    UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
-    UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
-
-    // Set inventory module for every pod with crew capacity.
-    DebugEx.Info("Loading pod inventories...");
-    foreach (AvailablePart avPart in PartLoader.LoadedPartsList) {
-      if (avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
-          || avPart.name == RdKerbalEva
-          || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1) {
-        continue;
+    int loadedPartCount;
+    int loadedPartIndex;
+
+    bool HasPodInventories (Part part)
+    {
+      for (int i = part.Modules.Count; i-- > 0; ) {
+        var inv = part.Modules[i] as ModuleKISInventory;
+        if (inv == null) {
+          continue;
+        }
+        if (inv.invType == ModuleKISInventory.InventoryType.Pod) {
+          return true;
+        }
       }
+      return false;
+    }
 
-      DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
-      for (int i = 0; i < avPart.partPrefab.CrewCapacity; i++) {
-        try {
-          var moduleInventory =
-            avPart.partPrefab.AddModule(typeof(ModuleKISInventory).Name) as ModuleKISInventory;
-          KIS_Shared.AwakePartModule(moduleInventory);
-          var baseFields = new BaseFieldList(moduleInventory);
-          baseFields.Load(nodeSettings.GetNode("EvaInventory"));
-          moduleInventory.podSeat = i;
-          moduleInventory.invType = ModuleKISInventory.InventoryType.Pod;
-          DebugEx.Fine("Pod inventory module(s) for seat {0} loaded successfully", i);
-        } catch {
-          DebugEx.Error("Pod inventory module(s) for seat {0} can't be loaded!", i);
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
+
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          if (HasPodInventories (avPart.partPrefab)) {
+            // pod inventories have already been added to the part by another
+            // mod, so skip the part (probably a dynamic crew capacity)
+			  DebugEx.Info("Skipping " + avPart.partPrefab);
+            continue;
+          }
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
         }
+        yield return null;
```

Comment:
As I mentioned before, I don't like an idea of having one part checked per a frame update. Some people play with FPS below 10, and there may be several hundreds of parts in the game (e.g. my game have over 700) - imagine the game loading speed now. It's definitely not worth it. If you absolutely want keeping this code, at least add the batching (handling 50-90 parts before returning).

Or, be even more smart and check the timings to figure out if returning the control back to the game loader makes sense. E.g. if you've handled 50 parts in just 20ms what is the point to bother about updating the progress bar? I'd say, it doesn't make sense to update the loading progress more frequently than 5-10 times per a frame - if you rise the frequency, a normal human won't catch the difference anyways.

Reply:
Oh, that's what you were referring to. OK.